### PR TITLE
ci: Build binaries on new tag.

### LINKED ISSUE #490: Provide binaries for every release in Github releases
Providing statically compiled binaries would make it much easier for users to simply try the software in the first place.
Since it has a lot of dependencies, compiling it from scratch is a pain on older systems. 
You might point towards distro packaging but having binaries helps with some immutable distros where packages are not feasible.

### PR DESCRIPTION
**Pull request by slowsage**
Resolves #490.

### CODE REVIEW

File: .github/workflows/binary.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          - {target: x86_64-unknown-linux-gnu, zipext: ".tar.gz"}
+          - {target: aarch64-unknown-linux-gnu, zipext: ".tar.gz"}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: cross
+      - run: cargo add openssl --features vendored
+      - run: cross build --locked --release --target=${{ matrix.platform.target }}
+      - run: echo "BINARY_NAME=$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[].targets[] | select( .kind | map(. == "bin") | any ) | .name')" >> $GITHUB_ENV
```
Comment (JakeStanger) [inline .github/workflows/binary.yml:24]: Can I ask that you add a blank newline between each step please for readability? 

Can steps also all have names please? It makes it a little cleaner in the logs than seeing a whole bash script for each section title.

**Review by JakeStanger — COMMENTED** [review]
Thanks for your work on this, greatly appreciated. Generally lgtm, just a few small bits to keep it tidy and food for thought.

I can see the job has ran (and failed) a few times in your fork. I can't see any releases over there though. Do you have any evidence of this working?

File: .github/workflows/binary.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      matrix:
+        platform:
+          - {target: x86_64-unknown-linux-gnu, zipext: ".tar.gz"}
+          - {target: aarch64-unknown-linux-gnu, zipext: ".tar.gz"}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: cross
+      - run: cargo add openssl --features vendored
```
[the remark this answers is not in the document]
Comment (slowsage) [reply .github/workflows/binary.yml:22]: Should be ok as long as it doesn't rebuild cross for each `cargo build` for devs. It has deps on docker and maybe more heavyweight than required for 99% of dev work. Never had a dev dep - not sure we want to buildi cross when a global `cargo binstall` will do. Either way, lmk what you prefer.

File: Cross.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[build]
+pre-build = [
+  "dpkg --add-architecture $CROSS_DEB_ARCH",
+  "apt-get update && apt-get install --assume-yes libssl-dev:$CROSS_DEB_ARCH libgtk-3-dev:$CROSS_DEB_ARCH libgtk-layer-shell-dev:$CROSS_DEB_ARCH libpulse-dev:$CROSS_DEB_ARCH",
```
[the remark this answers is not in the document]
Comment (slowsage) [reply Cross.toml:4]: Agreed. The github cli works well. Was thinking maybe an [env variable](https://docs.github.com/en/actions/learn-github-actions/variables).

File: .github/workflows/binary.yml (hunk 1 above)
Comment (slowsage) [reply .github/workflows/binary.yml:24]: Will do!

**Comment by slowsage** [comment]
> Thanks for your work on this, greatly appreciated. Generally lgtm, just a few small bits to keep it tidy and food for thought.
> 
> I can see the job has ran (and failed) a few times in your fork. I can't see any releases over there though. Do you have any evidence of this working?

Just created a test release earlier today - please take a look.

**Comment by slowsage** [comment]
Release `v0.14.3` on my repo shows a successful build with the `ubuntu_setup.sh` working. Verified deploy.yml is successfully installing packages with the new script (though it fails due to lack of access to crates.io).
The rust docs says dev dependencies are for tests and benchmarks. I don't know how to use cross as a dev dep in this workflow. Can you take this? My preference is to leave as is and deal with issues if they arise.
I've left it as 2 commits instead of squishing since both commits work and they are separable logically.

**Review by JakeStanger — APPROVED** [review]
This looks good. Script is a definite improvement so many thanks for that. I've tested a binary from your release and it runs and looks sensible, so all good on that front. Thanks for creating the test release.

Your thinking around the `openssl` crate sounds reasonable, so let's leave it for now.

**Comment by JakeStanger** [comment]
Unfortunately...it looks like the on release trigger only fires on *manual* release, which means when the workflow created the release the binary workflow did not trigger :(

Apparently it should be possible to add a `workflow_run` trigger, but I think this will require some adjustments to the workflow.

@slowsage is there any chance you'd be able to look into this please? No worries at all if not, appreciate this PR was a while ago now.

**Comment by slowsage** [comment]
Yes. I think I have a way to make it more robust in general (including workflow_dispatch as backup, will also leave on manual release).

**Pull request merged**